cocos/core/event/async-delegate.ts

```diff
@@ -0,0 +1,27 @@
+export class AsyncDelegate<T extends (...args: any) => (Promise<void> | void) = () => (Promise<void> | void)> {
+    private _delegates: T[] = [];
+
+    public add (callback: T) {
+        if (this._delegates.indexOf(callback) === -1) {
+            this._delegates.push(callback);
+        }
+    }
+
+    public hasListener (callback: T) {
+        if (this._delegates.indexOf(callback) !== -1) {
+            return true;
+        }
+        return false;
```

Comment:
```suggestion
        return this._delegates.includes(callback);
```

It's safe to do this.